specification/mcpp/resource-manager/Microsoft.CloudPartnerProgramMembership/preview/2023-02-09-preview/mcpp.json

```diff
@@ -0,0 +1,447 @@
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "List the CloudPartnerProgramMembership purchase details .",
+            "schema": {
+              "$ref": "#/definitions/McppPurchaseDetailsResponse"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "delete": {
+        "tags": ["CloudPartnerProgramMemberships"],
+        "operationId": "CloudPartnerProgramMemberships_CancelMcppPuchase",
+        "x-ms-examples": {
+          "deleteOperations": { "$ref": "./examples/cancelMcppPurchase.json" }
+        },
+        "description": "The operation to cancel a CloudPartnerProgramMembership offer",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Canceled the purchase"
+          },
+          "204": {
+            "description": "purchase record does not exist"
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "McppPurchaseDetailsResponse":{
+      "description": "MCPP Purchase details response",
+      "type": "object",
+      "x-ms-azure-resource": true,
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "id for the purchase resource",
+          "readOnly": true
+        },
+        "name": {
+          "description": "purchase resource name.",
+          "readOnly": true,
+          "type": "string"
+        },
+        "type": {
+          "description": "Resource type, CloudPartnerProgramMembership.",
+          "readOnly": true,
+          "type": "string"
+        },
+        "properties": { 
+          "$ref": "#/definitions/McppPurchaseDetailsResponseProperties",
+          "description": "MCPP Purchase details response properties."
+        }
+      }
+    },
+    "McppPurchaseDetailsResponseProperties": {
+      "description": "MCPP Purchase details response properties",
+      "type": "object",
+      "properties": {
+        "orderNo": {
+          "type": "string",
+          "description": "order number for the purchase",
+          "readOnly": true
+        },
+        "orderDateTime": {
+          "type": "string",
+          "description": "date time at which the order was placed",
```

Comment:
This sounds redundant with systemData.createdAt (tracked by ARM), assuming this represents the time when the cusomer submitted the order... If this represents the time the order was fulfilled/charged, then we should clarify that in this description, so the customer knows why this is different from systemData.createdAt.

Reply:
yes please leverage the latest common types for proxyresource and you will get systemdata definition included in that

Reply:
This indicates the time order was placed. Will systemData.createdAt be updated even if we are not onboarded to rpaas ? Or is this a property which our RP needs to maintain.

Reply:
when we move to rpass we would remove orderDateTime and get this from rpaas, until then this can stay.